### PR TITLE
Call the ClusterListener APIs asynchronously

### PR DESCRIPTION
**Pull request by adityadani**
**What this PR does / why we need it**:
- In the updateClusterStatus routine we continuously post the Node updates to all the cluster
  listeners. However they are being done synchronously .
- As in this routine the error from the Listener APIs is not processed, they can be called asynchronously.

**Special notes for your reviewer**:

Signed-off-by: Aditya Dani <aditya@portworx.com>

### CODE REVIEW

**Comment by sangleganesh** [comment]
with this change, more than one update can run in parallel, right ? you probably want to queue them up and process them in order, asynchronously (pcqueue with size = 1?)

**Comment by adityadani** [comment]
@sangleganesh I have updated the PR.

File: cluster/manager/manager.go [hunk 1]
```diff
@@ -84,14 +87,18 @@ func Init(cfg config.ClusterConfig) error {
...
 	}
-
 	inst = &ClusterManager{
 		listeners:    list.New(),
 		config:       cfg,
 		kv:           kv,
 		nodeCache:    make(map[string]api.Node),
 		nodeStatuses: make(map[string]api.Status),
 	}
+	if jobsched.Instance() == nil {
```
Comment (sangleganesh) [inline cluster/manager/manager.go]: nit: not needed anymore

File: pkg/pcqueue/pcqueue.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		cv:    sync.NewCond(mtx),
+		taskQ: list.New()}
+	workerFn := func() { q.dequeue() }
+	for i := uint8(0); i < numWorkers; i++ {
+		go workerFn()
+	}
+	return q
+}
+
+func (w *taskQueue) dequeue() {
```
Comment (lpabon) [inline pkg/pcqueue/pcqueue.go]: Why do this and not use channels? Is there a benefit?

**Pull request closed without merging**